addons/hr/models/hr_employee.py

```diff
@@ -284,6 +284,8 @@ def _sync_user(self, user, employee_has_image=False):
 
     @api.model
     def create(self, vals):
+        if vals.get('birthday') and self._isDateValid(vals['birthday']):
+            vals['birthday'] = ''
```

Comment:
this is not the good place to do it, probably, it should be somewhere in hr_contract_salary controllers, where you receive the information from the salary configurator.  And that's a place where we should filter out the buggy personal into